src/assets/TokensController.ts

```diff
@@ -216,32 +238,48 @@ export class TokensController extends BaseController<
     const releaseLock = await this.mutex.acquire();
     try {
       address = toChecksumHexAddress(address);
-      const { tokens, ignoredTokens } = this.state;
+      const { tokens, ignoredTokens, detectedTokens } = this.state;
+      let newTokens: Token[] = [...tokens];
       const isERC721 = await this._detectIsERC721(address);
-      const newEntry: Token = { address, symbol, decimals, image, isERC721 };
-      const previousEntry = tokens.find(
+      const tokenMetadata = await this.fetchTokenMetadata(address);
```

Comment:
Could this call be made in parallel with `_detectIsERC721`? They don't appear to depend on each other.

Reply:
Updated to use Promise.all